numpy/lib/tests/test_format.py

```diff
@@ -728,13 +728,15 @@ def test_read_magic():
 def test_read_magic_bad_magic():
     for magic in malformed_magic:
         f = BytesIO(magic)
-        yield raises(ValueError)(format.read_magic), f
+        with assert_raises(ValueError):
+            format.read_array(f)
```

Comment:
Convention seems to be to use the shorter `assert_raises(ValueError, format.read_array, f)` in almost all our test cases